### PR TITLE
feat(ci): Hash project euler solutions

### PR DESCRIPTION
**Pull request by maltejur**
### **Describe your change:**

As requested from the Project Euler Team and already discussed in the discord, we should hash the values of the Project Euler solutions for testing so that no direct solutions are stored. This pr hashes all the values in the `project_euler_answers.json` and updates the tests to use the hashed output.

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
~~This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.~~
~~All new Python files are placed inside an existing directory.~~
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
~~All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.~~
~~All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.~~
~~If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.~~

### CODE REVIEW

**Comment by cclauss** [comment]
As we have discussed elsewhere, I do not believe that Project Euler’s Community Commons license gives them the right to limit the use of their content in this way.

**Comment by maltejur** [comment]
Yes I agree that they can't force this on us with the CC license, but since they just want the solutions in the JSON file for testing hashed, so people won't just copy-paste everything from there, I don't see why we shouldn't do that. Is there any disadvantage in doing this?

**Comment by cclauss** [comment]
> please check the Discord

We are an open source project.  I am uncomfortable making decisions behind closed doors.  The Euler Project’s Community Commons license does not give them the right to limit our use of their IP except as expressed in that license.  If they want to change their license they can do so in the customary way.

**Review by cclauss — DISMISSED** [review]
Tests are failing.

File: scripts/validate_solutions.py [hunk 1]
```diff
@@ -89,5 +90,7 @@ def test_project_euler(solution_path: pathlib.Path) -> None:
     problem_number: str = solution_path.parent.name[8:].zfill(3)
     expected: str = PROBLEM_ANSWERS[problem_number]
     solution_module = convert_path_to_module(solution_path)
-    answer = str(solution_module.solution())  # type: ignore
-    assert answer == expected, f"Expected {expected} but got {answer}"
+    answer = hashlib.sha256(bytes(str(solution_module.solution()), "utf-8")).hexdigest() # type: ignore
```
Comment (siriak) [inline scripts/validate_solutions.py:93]:
```suggestion
    answer = hashlib.sha256(bytes(str(solution_module.solution()), "utf-8")).hexdigest()  # type: ignore
```

**Comment by siriak** [comment]
@maltejur build is failing on solutions validation, it seems that hashes in the JSON file are incorrect. Could you please check what's wrong?

**Review by Panquesito7 — APPROVED** [review]
Good work, @maltejur! Thank you! 🎉 👍

**Comment by cclauss** [comment]
@maltejur Thanks for doing this.

**Pull request merged**